iped-engine/src/main/java/iped/engine/task/index/IndexItem.java

```diff
@@ -196,16 +214,19 @@ public static boolean isRealNumber(String field) {
         return isFloat(field) || isDouble(field);
     }
 
-    public static Map<String, Class> getMetadataTypes() {
+    public static Map<String, Class<?>> getMetadataTypes() {
         return Collections.unmodifiableMap(typesMap);
     }
 
     public static void saveMetadataTypes(File confDir) throws IOException {
         File metadataTypesFile = new File(confDir, attrTypesFilename);
         UTF8Properties props = new UTF8Properties();
-        for (Object o : typesMap.entrySet().toArray()) {
-            Entry<String, Class<?>> e = (Entry<String, Class<?>>) o;
-            props.setProperty(e.getKey(), e.getValue().getCanonicalName());
+        for (Entry<String, Class<?>> e : typesMap.entrySet()) {
```

Comment:
Please put back the .toArray() in the for loop statement, it fixes a ConcurrentModificationException, see #942.

Reply:
Thanks!